build/build.go

```diff
@@ -579,6 +579,13 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 		}
 	}
 
+	// Propagate SOURCE_DATE_EPOCH from the client env
+	if v, ok := os.LookupEnv("SOURCE_DATE_EPOCH"); ok {
```

Comment:
I think this should be `ok && v != ""` as `SOURCE_DATE_EPOCH=` could be a common way to reset the behavior.

Reply:
To support `SOURCE_DATE_EPOCH=`, the condition cannot contain `v != ""` IIUC

Reply:
wdym? I'm thinking of the case where environment has `SOURCE_DATE_EPOCH` defined but the user doesn't want to forward it to the buildkit(user wants buildkit to just use the current time). What should they set?

Reply:
When the Dockerfile already has `ARG SOURCE_DATE_EPOCH=<non empty value>`, I'd expect `export SOURCE_DATE_EPOCH=` to set the build arg to an empty string.

Reply:
Dockerfile defining `SOURCE_DATE_EPOCH` is only one possibility. The user might just want to control the timestamps in the config metadata and not the `RUN` commands.

Reply:
fixed